packages/backend/src/modules/tvl2/modules/TvlCleaner.ts

```diff
@@ -0,0 +1,93 @@
+import { Logger } from '@l2beat/backend-tools'
+import { UnixTime } from '@l2beat/shared-pure'
+
+import { TvlCleanerRepository } from '../../../peripherals/database/TvlCleanerRepository'
+import { Clock } from '../../../tools/Clock'
+import { TaskQueue } from '../../../tools/queue/TaskQueue'
+export interface TvlRepositoryToClean {
+  deleteHourlyUntil: (dateRange: {
+    from: UnixTime | undefined
+    to: UnixTime
+  }) => Promise<number>
+  deleteSixHourlyUntil: (dateRange: {
+    from: UnixTime | undefined
+    to: UnixTime
+  }) => Promise<number>
+}
+
+/**
+ * WARNING: Not all updaters support cleaning, so their repositories should not be injected there.
+ * Current list of unsupported updaters: PriceUpdater, CirculatingSupplyUpdater
+ */
+export class TvlCleaner {
+  private readonly taskQueue: TaskQueue<void>
+
+  constructor(
+    private readonly clock: Clock,
+    private readonly logger: Logger,
+    private readonly tvlCleanerRepository: TvlCleanerRepository,
+    private readonly repositoriesToClean: TvlRepositoryToClean[],
+  ) {
+    this.logger = this.logger.for(this)
+    this.taskQueue = new TaskQueue(
+      () => this.clean(),
+      this.logger.for('taskQueue'),
+      {
+        metricsId: TvlCleaner.name,
+      },
+    )
+  }
+
+  start() {
+    this.taskQueue.addToFront()
+    this.logger.info('Started')
+    return this.clock.onNewHour(() => {
+      this.taskQueue.addToFront()
+    })
+  }
+
+  async clean() {
+    const hourlyDeletionBoundary =
+      this.clock._TVL_ONLY_getHourlyDeletionBoundary()
```

Comment:
those methods will be phased out from the codebase - instead inject SyncOptimizer to get the boundaries